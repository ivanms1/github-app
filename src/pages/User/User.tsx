import React from "react";
import { Text } from "@mantine/core";
import { motion } from "framer-motion";

import RepositoryList from "@/components/RepositoryList";

import {
  SearchRepositoriesQuery,
  useUserStarredRepositoriesQuery,
} from "src/generated/graphql";

import styles from "./User.module.scss";

function User() {
  const { data, loading, fetchMore } = useUserStarredRepositoriesQuery({
    notifyOnNetworkStatusChange: true,
  });

  const onRefetch = () => {
    if (!data?.viewer?.starredRepositories.pageInfo?.hasNextPage) {
      return;
    }

    fetchMore({
      variables: {
        after: data?.viewer.starredRepositories?.pageInfo?.endCursor,
      },
    });
  };

  return (
    <motion.div
      initial={{ x: -500 }}
      animate={{ x: 0 }}
      transition={{
        type: "spring",
        damping: 12,
        stiffness: 100,
      }}
    >
      <Text className={styles.Title} weight="bold">
        My starred repositories
      </Text>
      <RepositoryList
        repositories={
          data?.viewer?.starredRepositories.repositories?.map(
            (repo) => repo?.node
          ) as SearchRepositoriesQuery["search"]["repositories"]
        }
        loading={loading}
        cursor={data?.viewer?.starredRepositories?.pageInfo?.endCursor}
        onRefetch={onRefetch}
      />
    </motion.div>
  );
}

export default User;
