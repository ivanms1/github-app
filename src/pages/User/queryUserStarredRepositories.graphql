query UserStarredRepositories($after: String) {
  viewer {
    login
    name
    starredRepositories(first: 10, after: $after) {
      pageInfo {
        startCursor
        hasNextPage
        endCursor
      }
      repositories: edges {
        node {
          id
          name
          nameWithOwner
          description
          updatedAt
          createdAt
          pushedAt
          stargazerCount
          viewerHasStarred
          primaryLanguage {
            color
            name
            id
          }
        }
      }
    }
  }
}
