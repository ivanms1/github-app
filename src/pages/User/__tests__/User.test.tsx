import { render } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";

import User from "../User";

import { UserStarredRepositoriesDocument } from "src/generated/graphql";

import data from "__mocks__/user-starred-repositories";

const mocks: any = [
  {
    request: {
      query: UserStarredRepositoriesDocument,
    },
    result: data,
  },
];

describe("User page", () => {
  beforeEach(() => {
    // Reduce animations so snapshots are consistent
    jest.mock("framer-motion", () => ({
      ...jest.requireActual("framer-motion"),
      useReducedMotion: () => true,
    }));

    // stub date, so relative time is consistent
    Date.now = jest.fn(() => new Date(Date.UTC(2017, 7, 9, 8)).valueOf());
  });

  test("it should render the user starred repositories", async () => {
    const { container } = render(
      <MockedProvider mocks={mocks}>
        <User />
      </MockedProvider>
    );

    // wait for data to load
    await new Promise((resolve) => setTimeout(resolve, 100));
    expect(container).toMatchSnapshot();
  });
});
