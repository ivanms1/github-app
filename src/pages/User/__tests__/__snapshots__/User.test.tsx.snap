// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User page it should render the user starred repositories 1`] = `
<div>
  <div
    style="transform: translateX(-436.8368177947951px) translateZ(0);"
  >
    <div
      class="mantine-Text-root Title mantine-1xctzl"
    >
      My starred repositories
    </div>
    <div
      class="Container"
      style="opacity: 1; transform: scale(0) translateZ(0);"
    >
      <symbol
        class="mantine-Paper-root mantine-Card-root RepositoryCard mantine-1lay5d4"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <a
          href="https://github.com/freeCodeCamp/freeCodeCamp"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="mantine-Text-root Title mantine-14hwddk"
          >
            freeCodeCamp/freeCodeCamp
          </div>
        </a>
        <div
          class="mantine-Text-root mantine-2qa0ve"
        >
          freeCodeCamp.org's open-source codebase and curriculum. Learn to code for free.
        </div>
        <div
          class="InfoBox"
        >
          <div
            class="mantine-Badge-light mantine-Badge-root mantine-1v8afah"
          >
            <span
              class="mantine-h9iq4m mantine-Badge-inner"
            >
              TypeScript
            </span>
          </div>
          <div
            class="StarBox"
          >
            <button
              class="mantine-UnstyledButton-root mantine-14iebtj"
              type="button"
            >
              <svgrurl />
            </button>
             
            <div
              class="mantine-Text-root mantine-1f83q5f"
            >
              348357
            </div>
          </div>
          <div
            class="mantine-Text-root mantine-1f83q5f"
          >
            in 5 years
          </div>
        </div>
      </symbol>
      <symbol
        class="mantine-Paper-root mantine-Card-root RepositoryCard mantine-1lay5d4"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <a
          href="https://github.com/chingu-voyage5/Geckos-Team-43"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="mantine-Text-root Title mantine-14hwddk"
          >
            chingu-voyage5/Geckos-Team-43
          </div>
        </a>
        <div
          class="mantine-Text-root mantine-2qa0ve"
        >
          Add-project-description-here | Voyage-5 | chingu.io
        </div>
        <div
          class="InfoBox"
        >
          <div
            class="mantine-Badge-light mantine-Badge-root mantine-12tfcnq"
          >
            <span
              class="mantine-h9iq4m mantine-Badge-inner"
            >
              JavaScript
            </span>
          </div>
          <div
            class="StarBox"
          >
            <button
              class="mantine-UnstyledButton-root mantine-14iebtj"
              type="button"
            >
              <svgrurl />
            </button>
             
            <div
              class="mantine-Text-root mantine-1f83q5f"
            >
              3
            </div>
          </div>
          <div
            class="mantine-Text-root mantine-1f83q5f"
          >
            in 1 year
          </div>
        </div>
      </symbol>
      <symbol
        class="mantine-Paper-root mantine-Card-root RepositoryCard mantine-1lay5d4"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <a
          href="https://github.com/hackerbayuniversity/Course"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="mantine-Text-root Title mantine-14hwddk"
          >
            hackerbayuniversity/Course
          </div>
        </a>
        <div
          class="mantine-Text-root mantine-2qa0ve"
        >
          Open Source course materials from HackerBay University
        </div>
        <div
          class="InfoBox"
        >
          <div
            class="mantine-Badge-light mantine-Badge-root mantine-12tfcnq"
          >
            <span
              class="mantine-h9iq4m mantine-Badge-inner"
            >
              JavaScript
            </span>
          </div>
          <div
            class="StarBox"
          >
            <button
              class="mantine-UnstyledButton-root mantine-14iebtj"
              type="button"
            >
              <svgrurl />
            </button>
             
            <div
              class="mantine-Text-root mantine-1f83q5f"
            >
              68
            </div>
          </div>
          <div
            class="mantine-Text-root mantine-1f83q5f"
          >
            in 1 year
          </div>
        </div>
      </symbol>
      <symbol
        class="mantine-Paper-root mantine-Card-root RepositoryCard mantine-1lay5d4"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <a
          href="https://github.com/chingu-voyage7/Bears-Team-29"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="mantine-Text-root Title mantine-14hwddk"
          >
            chingu-voyage7/Bears-Team-29
          </div>
        </a>
        <div
          class="mantine-Text-root mantine-2qa0ve"
        >
          eCommerce site | Voyage-7 | https://chingu.io/
        </div>
        <div
          class="InfoBox"
        >
          <div
            class="mantine-Badge-light mantine-Badge-root mantine-12tfcnq"
          >
            <span
              class="mantine-h9iq4m mantine-Badge-inner"
            >
              JavaScript
            </span>
          </div>
          <div
            class="StarBox"
          >
            <button
              class="mantine-UnstyledButton-root mantine-14iebtj"
              type="button"
            >
              <svgrurl />
            </button>
             
            <div
              class="mantine-Text-root mantine-1f83q5f"
            >
              1
            </div>
          </div>
          <div
            class="mantine-Text-root mantine-1f83q5f"
          >
            in 1 year
          </div>
        </div>
      </symbol>
      <symbol
        class="mantine-Paper-root mantine-Card-root RepositoryCard mantine-1lay5d4"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <a
          href="https://github.com/lingui/js-lingui"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="mantine-Text-root Title mantine-14hwddk"
          >
            lingui/js-lingui
          </div>
        </a>
        <div
          class="mantine-Text-root mantine-2qa0ve"
        >
          üåçüìñ A readable, automated, and optimized (5 kb) internationalization for JavaScript
        </div>
        <div
          class="InfoBox"
        >
          <div
            class="mantine-Badge-light mantine-Badge-root mantine-1v8afah"
          >
            <span
              class="mantine-h9iq4m mantine-Badge-inner"
            >
              TypeScript
            </span>
          </div>
          <div
            class="StarBox"
          >
            <button
              class="mantine-UnstyledButton-root mantine-14iebtj"
              type="button"
            >
              <svgrurl />
            </button>
             
            <div
              class="mantine-Text-root mantine-1f83q5f"
            >
              3384
            </div>
          </div>
          <div
            class="mantine-Text-root mantine-1f83q5f"
          >
            in 5 years
          </div>
        </div>
      </symbol>
      <symbol
        class="mantine-Paper-root mantine-Card-root RepositoryCard mantine-1lay5d4"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <a
          href="https://github.com/facebook/react"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="mantine-Text-root Title mantine-14hwddk"
          >
            facebook/react
          </div>
        </a>
        <div
          class="mantine-Text-root mantine-2qa0ve"
        >
          A declarative, efficient, and flexible JavaScript library for building user interfaces.
        </div>
        <div
          class="InfoBox"
        >
          <div
            class="mantine-Badge-light mantine-Badge-root mantine-12tfcnq"
          >
            <span
              class="mantine-h9iq4m mantine-Badge-inner"
            >
              JavaScript
            </span>
          </div>
          <div
            class="StarBox"
          >
            <button
              class="mantine-UnstyledButton-root mantine-14iebtj"
              type="button"
            >
              <svgrurl />
            </button>
             
            <div
              class="mantine-Text-root mantine-1f83q5f"
            >
              190753
            </div>
          </div>
          <div
            class="mantine-Text-root mantine-1f83q5f"
          >
            in 5 years
          </div>
        </div>
      </symbol>
      <symbol
        class="mantine-Paper-root mantine-Card-root RepositoryCard mantine-1lay5d4"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <a
          href="https://github.com/prevwong/craft.js"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="mantine-Text-root Title mantine-14hwddk"
          >
            prevwong/craft.js
          </div>
        </a>
        <div
          class="mantine-Text-root mantine-2qa0ve"
        >
          üöÄ A React Framework for building extensible drag and drop page editors
        </div>
        <div
          class="InfoBox"
        >
          <div
            class="mantine-Badge-light mantine-Badge-root mantine-1v8afah"
          >
            <span
              class="mantine-h9iq4m mantine-Badge-inner"
            >
              TypeScript
            </span>
          </div>
          <div
            class="StarBox"
          >
            <button
              class="mantine-UnstyledButton-root mantine-14iebtj"
              type="button"
            >
              <svgrurl />
            </button>
             
            <div
              class="mantine-Text-root mantine-1f83q5f"
            >
              5024
            </div>
          </div>
          <div
            class="mantine-Text-root mantine-1f83q5f"
          >
            in 5 years
          </div>
        </div>
      </symbol>
      <symbol
        class="mantine-Paper-root mantine-Card-root RepositoryCard mantine-1lay5d4"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <a
          href="https://github.com/ivanms1/nextjs-graphql-typescript"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="mantine-Text-root Title mantine-14hwddk"
          >
            ivanms1/nextjs-graphql-typescript
          </div>
        </a>
        <div
          class="mantine-Text-root mantine-2qa0ve"
        >
          GraphQL + TypeScript boilerplate for NextJS.
        </div>
        <div
          class="InfoBox"
        >
          <div
            class="mantine-Badge-light mantine-Badge-root mantine-1v8afah"
          >
            <span
              class="mantine-h9iq4m mantine-Badge-inner"
            >
              TypeScript
            </span>
          </div>
          <div
            class="StarBox"
          >
            <button
              class="mantine-UnstyledButton-root mantine-14iebtj"
              type="button"
            >
              <svgrurl />
            </button>
             
            <div
              class="mantine-Text-root mantine-1f83q5f"
            >
              7
            </div>
          </div>
          <div
            class="mantine-Text-root mantine-1f83q5f"
          >
            in 5 years
          </div>
        </div>
      </symbol>
      <symbol
        class="mantine-Paper-root mantine-Card-root RepositoryCard mantine-1lay5d4"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <a
          href="https://github.com/ivanms1/next-js-template"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="mantine-Text-root Title mantine-14hwddk"
          >
            ivanms1/next-js-template
          </div>
        </a>
        <div
          class="mantine-Text-root mantine-2qa0ve"
        >
          Next.js template
        </div>
        <div
          class="InfoBox"
        >
          <div
            class="mantine-Badge-light mantine-Badge-root mantine-1v8afah"
          >
            <span
              class="mantine-h9iq4m mantine-Badge-inner"
            >
              TypeScript
            </span>
          </div>
          <div
            class="StarBox"
          >
            <button
              class="mantine-UnstyledButton-root mantine-14iebtj"
              type="button"
            >
              <svgrurl />
            </button>
             
            <div
              class="mantine-Text-root mantine-1f83q5f"
            >
              2
            </div>
          </div>
          <div
            class="mantine-Text-root mantine-1f83q5f"
          >
            in 5 years
          </div>
        </div>
      </symbol>
      <symbol
        class="mantine-Paper-root mantine-Card-root RepositoryCard mantine-1lay5d4"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <a
          href="https://github.com/project-shelf/project-shelf"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="mantine-Text-root Title mantine-14hwddk"
          >
            project-shelf/project-shelf
          </div>
        </a>
        <div
          class="mantine-Text-root mantine-2qa0ve"
        >
          A showcase app for your projects
        </div>
        <div
          class="InfoBox"
        >
          <div
            class="mantine-Badge-light mantine-Badge-root mantine-1v8afah"
          >
            <span
              class="mantine-h9iq4m mantine-Badge-inner"
            >
              TypeScript
            </span>
          </div>
          <div
            class="StarBox"
          >
            <button
              class="mantine-UnstyledButton-root mantine-14iebtj"
              type="button"
            >
              <svgrurl />
            </button>
             
            <div
              class="mantine-Text-root mantine-1f83q5f"
            >
              11
            </div>
          </div>
          <div
            class="mantine-Text-root mantine-1f83q5f"
          >
            in 5 years
          </div>
        </div>
      </symbol>
      <symbol
        class="mantine-Paper-root mantine-Card-root RepositoryCard mantine-1lay5d4"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <a
          href="https://github.com/meetncode/meetncode"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="mantine-Text-root Title mantine-14hwddk"
          >
            meetncode/meetncode
          </div>
        </a>
        <div
          class="mantine-Text-root mantine-2qa0ve"
        >
          Meetups for developers
        </div>
        <div
          class="InfoBox"
        >
          <div
            class="mantine-Badge-light mantine-Badge-root mantine-12tfcnq"
          >
            <span
              class="mantine-h9iq4m mantine-Badge-inner"
            >
              JavaScript
            </span>
          </div>
          <div
            class="StarBox"
          >
            <button
              class="mantine-UnstyledButton-root mantine-14iebtj"
              type="button"
            >
              <svgrurl />
            </button>
             
            <div
              class="mantine-Text-root mantine-1f83q5f"
            >
              2
            </div>
          </div>
          <div
            class="mantine-Text-root mantine-1f83q5f"
          >
            in 5 years
          </div>
        </div>
      </symbol>
      <symbol
        class="mantine-Paper-root mantine-Card-root RepositoryCard mantine-1lay5d4"
        style="opacity: 0; transform: translateY(20px) translateZ(0);"
        variants="[object Object]"
        whilehover="[object Object]"
      >
        <a
          href="https://github.com/florinpop17/weekly-projects"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="mantine-Text-root Title mantine-14hwddk"
          >
            florinpop17/weekly-projects
          </div>
        </a>
        <div
          class="mantine-Text-root mantine-2qa0ve"
        >
          Build a project every week to improve your coding skills
        </div>
        <div
          class="InfoBox"
        >
          <div
            class="StarBox"
          >
            <button
              class="mantine-UnstyledButton-root mantine-14iebtj"
              type="button"
            >
              <svgrurl />
            </button>
             
            <div
              class="mantine-Text-root mantine-1f83q5f"
            >
              420
            </div>
          </div>
          <div
            class="mantine-Text-root mantine-1f83q5f"
          >
            in 3 years
          </div>
        </div>
      </symbol>
    </div>
    <span
      style="font-size: 0px;"
    />
  </div>
</div>
`;
